File: src/main/java/com/overclock/overclock/service/TestService.java
package com.overclock.overclock.service;

import com.overclock.overclock.model.Assembly;
import com.overclock.overclock.model.CPU;
import com.overclock.overclock.model.GPU;
import com.overclock.overclock.model.Overclock;
import com.overclock.overclock.model.RAM;
import com.overclock.overclock.model.enums.Program;

import java.math.BigDecimal;

public interface TestService {
    BigDecimal testAssembly(Assembly assembly);
    BigDecimal testAssemblyOnProgram(Assembly assembly, Program program);
    BigDecimal testCPU(CPU cpu, Overclock overclock);
    BigDecimal testGPU(GPU gpu, Overclock overclock);
    BigDecimal testRAM(RAM ram, Overclock overclock);
}

File: src/main/resources/frontend/angular/src/app/service/test.service.ts
import { Injectable } from '@angular/core';
import {environment} from "../../environments/environment";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";

@Injectable({
  providedIn: 'root'
})
export class TestService {
  private apiServerUrl = environment.apiBaseUrl;

  constructor(private http: HttpClient) { }

  public testAssembly(assemblyId: number): Observable<number> {
    return this.http.get<number>(`${this.apiServerUrl}/test/assembly/${assemblyId}`);
  }

  public testAssemblyOnProgram(assemblyId: number, program: string): Observable<number> {
    return this.http.get<number>(`${this.apiServerUrl}/test/assembly/${assemblyId}/${program}`);
  }

  public testCPU(assemblyId: number): Observable<number> {
    return this.http.get<number>(`${this.apiServerUrl}/test/assembly/${assemblyId}/component/cpu`);
  }

  public testGPU(assemblyId: number): Observable<number> {
    return this.http.get<number>(`${this.apiServerUrl}/test/assembly/${assemblyId}/component/gpu`);
  }

  public testRAM(assemblyId: number): Observable<number> {
    return this.http.get<number>(`${this.apiServerUrl}/test/assembly/${assemblyId}/component/ram`);
  }
}
